### PR TITLE
User role entities

### PR DESCRIPTION
**Pull request by Bozhokmaria**

### CODE REVIEW

File: src/main/java/com/exadel/discountwebapp/user/entity/User.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    @Column(name = "usr_id")
+    private Long id;
+
+    @Column(name = "usr_name", length = 50, nullable = false)
+    private String name;
+
+    @Column(name = "usr_surname", length = 50, nullable = false)
+    private String surname;
+
+    @Column(name = "usr_email", length = 100, unique = true, nullable = false)
```
Comment (jioter) [inline src/main/java/com/exadel/discountwebapp/user/entity/User.java]: look at schema length constraint for email 255

File: src/main/java/com/exadel/discountwebapp/user/entity/User.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    private String email;
+
+    @EqualsAndHashCode.Exclude
+    @Column(name = "usr_password", length = 255, nullable = false)
```
Comment (jioter) [inline src/main/java/com/exadel/discountwebapp/user/entity/User.java]: there is also new user password constraint length 510

**Review by jioter — DISMISSED** [review]

File: src/main/java/com/exadel/discountwebapp/user/entity/User.java (hunk 1 above)
Comment (Bozhokmaria) [reply src/main/java/com/exadel/discountwebapp/user/entity/User.java]: corrected

File: src/main/java/com/exadel/discountwebapp/user/entity/User.java (hunk 2 above)
Comment (Bozhokmaria) [reply src/main/java/com/exadel/discountwebapp/user/entity/User.java]: done

File: src/main/java/com/exadel/discountwebapp/user/entity/User.java [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    private Location location;
+
+    @ManyToOne
+    @EqualsAndHashCode.Exclude
+    @JoinColumn(name = "rol_id")
+    private Role role;
+
+    @OneToMany(mappedBy = "user")
+    @EqualsAndHashCode.Exclude
+    private List<UserDiscount> discounts = new ArrayList<>();
```
Comment (ahlukhouski2) [inline src/main/java/com/exadel/discountwebapp/user/entity/User.java:55]: UserDiscount? cannot find it in the PR

File: src/main/java/com/exadel/discountwebapp/user/entity/User.java [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+    @Column(name = "usr_name", length = 50, nullable = false)
+    private String name;
+
+    @Column(name = "usr_surname", length = 50, nullable = false)
+    private String surname;
+
+    @Column(name = "usr_email", length = 255, unique = true, nullable = false)
+    private String email;
+
+    @EqualsAndHashCode.Exclude
```
Comment (ahlukhouski2) [inline src/main/java/com/exadel/discountwebapp/user/entity/User.java]: password is a part of user identity, so we should not exclude it from equals/hashcode

File: src/main/java/com/exadel/discountwebapp/role/entity/Role.java [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    @EqualsAndHashCode.Exclude
+    @Column(name = "rol_id")
+    private Long id;
+
+    @Column(name = "rol_name", length = 25, unique = true, nullable = false)
+    private String name;
+
+    @OneToMany(mappedBy = "role")
```
Comment (ahlukhouski2) [inline src/main/java/com/exadel/discountwebapp/role/entity/Role.java:34]: should be excluded from equals/hashcode and tostring

File: src/main/java/com/exadel/discountwebapp/user/entity/User.java [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+public class User {
+
+    @Id
+    @EqualsAndHashCode.Exclude
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    @Column(name = "usr_id")
+    private Long id;
+
+    @Column(name = "usr_name", length = 50, nullable = false)
+    private String name;
```
Comment (ahlukhouski2) [inline src/main/java/com/exadel/discountwebapp/user/entity/User.java]: lets replace with firstName (usr_first_name)

File: src/main/java/com/exadel/discountwebapp/user/entity/User.java [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+
+    @Column(name = "usr_surname", length = 50, nullable = false)
+    private String surname;
```
Comment (ahlukhouski2) [inline src/main/java/com/exadel/discountwebapp/user/entity/User.java]: lets replace with lastName (usr_last_name)

File: src/main/java/com/exadel/discountwebapp/userdiscount/entity/UserDiscount.java [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+    @Column(name = "ud_modified", nullable = false)
+    private LocalDateTime modified;
+
+    @Embeddable
+    @Data
+    @NoArgsConstructor
+    @AllArgsConstructor
+    public static class UserDiscountId implements Serializable {
+
+        private Long userId;
```
Comment (ahlukhouski2) [inline src/main/java/com/exadel/discountwebapp/userdiscount/entity/UserDiscount.java]: Lets add @Column( name = "usr_id" )

File: src/main/java/com/exadel/discountwebapp/userdiscount/entity/UserDiscount.java [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+
+        private Long discountId;
```
Comment (ahlukhouski2) [inline src/main/java/com/exadel/discountwebapp/userdiscount/entity/UserDiscount.java]: Lets add @column( name = "dis_id" )

File: src/main/java/com/exadel/discountwebapp/userdiscount/entity/UserDiscount.java [hunk 10]
```diff
@@ -0,0 +1,60 @@
...
+
+@Builder
+@Entity
+@Data
+@AllArgsConstructor
+@NoArgsConstructor
+@EqualsAndHashCode
+@Table(name = "user_discount")
+@EntityListeners(AuditingEntityListener.class)
+public class UserDiscount {
```
Comment (ahlukhouski2) [inline src/main/java/com/exadel/discountwebapp/userdiscount/entity/UserDiscount.java]: lets move UserDiscount to user package

**Review by ahlukhouski2 — APPROVED** [review]

**Review by VolodymyrBushko — APPROVED** [review]

**Pull request merged**